pkg/meta/meta.go

```diff
@@ -245,6 +258,23 @@ func SetExternalName(o metav1.Object, name string) {
 	AddAnnotations(o, map[string]string{AnnotationKeyExternalName: name})
 }
 
+// GetExternalCreateTime returns the time at which the external resource was
+// most recently created.
+func GetExternalCreateTime(o metav1.Object) *metav1.Time {
+	a := o.GetAnnotations()[AnnotationKeyExternalCreateTime]
+	t, err := time.Parse(time.RFC3339, a)
+	if err != nil {
+		return nil
+	}
+	return &metav1.Time{Time: t}
+}
+
+// SetExternalCreateTime sets the time at which the external resource was most
+// recently created to the current time.
+func SetExternalCreateTime(o metav1.Object) {
+	AddAnnotations(o, map[string]string{AnnotationKeyExternalCreateTime: metav1.Now().Format(time.RFC3339)})
+}
```

Comment:
I wonder whether this function should take the time as an argument, rather than assuming it to be the call time? I feel like in 99% of cases it would be called as `meta.SetExternalCreateTime(o, metav1.Now())` but perhaps that's not so bad for a little extra flexibility.

Reply:
I decided to update this to take the create time as an argument.